examples/qualcomm/oss_scripts/qwen/decoder_model_wrapper.py

```diff
@@ -0,0 +1,141 @@
+                "Please set `generation_config.cache_implementation='static'`."
+            )
+
+        self.model = model
+        self.static_cache = StaticCache(
+            config=self.model.config,
+            batch_size=self.model.generation_config.cache_config.batch_size,
+            max_cache_len=self.model.generation_config.cache_config.max_cache_len,
+            dtype=self.model.dtype,
+        )
+        for i in range(len(self.static_cache.key_cache)):
+            self.register_buffer(f"key_cache_{i}", self.static_cache.key_cache[i])
+            self.register_buffer(f"value_cache_{i}", self.static_cache.value_cache[i])
+        self.is_causal = any("CausalLM" in arch for arch in self.model.config.architectures)
+        # ====================Qualcomm Changed=================================
+        if self.is_causal:
+            mask = torch.full(
+                (1, 1, self.static_cache.max_cache_len, self.static_cache.max_cache_len),
+                float("-65504"),
+            )
+            mask = torch.triu(mask, diagonal=1)
+            self.register_buffer("mask", mask, persistent=False)
+
+        freqs_cos, freqs_sin = _get_freqs_cis(self.model.config)
+        self.register_buffer("freqs_cos", freqs_cos, persistent=False)
+        self.register_buffer("freqs_sin", freqs_sin, persistent=False)
+        # =====================================================================
+
+    def forward(self, input_ids: torch.Tensor, cache_position: torch.Tensor):
+        """
+        Forward pass of the module, which is compatible with the ExecuTorch runtime.
+
+        Args:
+            input_ids (`torch.Tensor`): Tensor representing current input token id to the module.
+            cache_position (`torch.Tensor`): Tensor representing current input position in the cache.
+
+        Returns:
+            torch.Tensor: Logits output from the model.
+
+        This forward adapter serves two primary purposes:
+
+        1. **Making the Model `torch.export`-Compatible**:
+            The adapter hides unsupported objects, such as the `Cache`, from the graph inputs and outputs,
+            enabling the model to be exportable using `torch.export` without encountering issues.
+
+        2. **Ensuring Compatibility with `ExecuTorch` runtime**:
+            The adapter matches the model's forward signature with that in `executorch/extension/llm/runner`,
+            ensuring that the exported model can be executed in `ExecuTorch` out-of-the-box.
+        """
+        # ====================Qualcomm Changed=================================
+        freqs_cos = self.freqs_cos[cache_position]
+        freqs_sin = self.freqs_sin[cache_position]
+
+        attn_mask = self.mask[:,:,cache_position, :] if self.is_causal else None
+        # =====================================================================
+        outs = self.model(
+            input_ids=input_ids,
+            attention_mask=attn_mask,
```

Comment:
One question: if you have to specify per layer mask, how would you?

@guangy10 does transformer api allow for per layer mask be specified here as list of tensor or something?